cql3/restrictions/statement_restrictions.cc

```diff
@@ -2748,14 +2749,27 @@ void statement_restrictions::prepare_indexed_global(const schema& idx_tbl_schema
         (*_idx_tbl_ck_prefix)[pos] = replace_column_def(e, &idx_tbl_schema.clustering_column_at(pos));
     }
 
+    if (std::ranges::any_of(*_idx_tbl_ck_prefix | std::views::drop(1), is_empty_restriction)) {
+        // If the partition key is not fully restricted, the index clustering key is of no use.
```

Comment:
The comment suggests this tests the (base table's) partition key, but doesn't idx_tbl_ck_prefix contain the token (which you drop()), then the partition columns, and then the clustering keys? In the next line you use take(partition_key_size()) - why isn't the same thing needed here too?

Reply:
At this point, the base table clustering keys haven't been added yet. That happens immediately after this if (if it doesn't return).

Reply:
Yes, I see now the add_clustering_restrictions_to_idx_ck_prefix one line below. I didn't see it because it was the only line this patch didn't change :-)